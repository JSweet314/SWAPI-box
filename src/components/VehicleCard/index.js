import React from 'react';
import PropTypes from 'prop-types';
import './style.css';

const VehicleCard = ({card, favorites, handleOnClick}) => {
  const selected = favorites.some(favorite => favorite.name === card.name) ?
    'selected' : '';

  const {name, model, vehicleClass, numberOfPassengers} = card;

  return (
    <article className="info-card">
      <div>
        <div className="info-card__top-line">
          <h3>{name}</h3>
          <button 
            onClick={() => handleOnClick(card, 'vehicles')}
            className={selected}></button>
        </div>
        <p>Model: {model}</p>
        <p>Class: {vehicleClass}</p>
        <p>Passengers: {numberOfPassengers}</p>
      </div>
    </article>
  );
};

VehicleCard.propTypes = {
  card: PropTypes.shape({
    name: PropTypes.string.isRequired,
    model: PropTypes.string.isRequired,
    vehicleClass: PropTypes.string.isRequired,
    numberOfPassengers: PropTypes.string.isRequired
  }).isRequired,
  favorites: PropTypes.array.isRequired,
  handleOnClick: PropTypes.func.isRequired
};

export default VehicleCard;