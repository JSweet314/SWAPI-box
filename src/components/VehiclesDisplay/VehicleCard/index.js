import React from 'react';
import './style.css';
import PropTypes from 'prop-types';

const VehicleCard = ({card, favorites}) => {
  const selected = favorites.some(favorite => favorite.name === card.name) ?
    'selected' : '';

  const {name, model, vehicleClass, numberOfPassengers} = card;

  return (
    <article className="info-card">
      <div>
        <div className="info-card__top-line">
          <h3>{name}</h3>
          <button className={selected}></button>
        </div>
        <p>Model: {model}</p>
        <p>Class: {vehicleClass}</p>
        <p>Passengers: {numberOfPassengers}</p>
      </div>
    </article>
  );
};

VehicleCard.propTypes = {
  card: PropTypes.shape({
    name: PropTypes.string.isRequired,
    model: PropTypes.string.isRequired,
    vehicleClass: PropTypes.string.isRequired,
    numberOfPassengers: PropTypes.string.isRequired
  }).isRequired,
  favorites: PropTypes.array.isRequired
};

export default VehicleCard;