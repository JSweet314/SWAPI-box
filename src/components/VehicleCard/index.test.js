import React from 'react';
import { shallow } from 'enzyme';
import VehicleCard from './index';

describe('VehicleCard', () => {
  /* eslint-disable no-undef */
  const mockHandleFavoriteClick = jest.fn();
  /* eslint-enable no-undef */

  const card = {
    name: 'Sand Crawler',
    model: 'Digger Crawler',
    vehicleClass: "wheeled",
    numberOfPassengers: '50'
  };

  it('should match a snapshot', () => {
    const wrapper = shallow(
      <VehicleCard
        handleFavoriteClick={mockHandleFavoriteClick}
        favorites={[]}
        card={card} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('should show as favorited if name matches obj in favorites array', () => {
    const wrapper = shallow(
      <VehicleCard
        handleFavoriteClick={mockHandleFavoriteClick}
        favorites={[{ name: 'Sand Crawler' }]}
        card={card} />
    );

    expect(wrapper.find('.selected').length).toEqual(1);
  });
});