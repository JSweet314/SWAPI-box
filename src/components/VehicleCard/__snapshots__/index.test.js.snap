// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleCard should match a snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <VehicleCard
    card={
        Object {
            "model": "Digger Crawler",
            "name": "Sand Crawler",
            "numberOfPassengers": "50",
            "vehicleClass": "wheeled",
          }
    }
    favorites={Array []}
    handleFavoriteClick={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <div
                className="info-card__top-line"
        >
                <h3>
                        Sand Crawler
                </h3>
                <button
                        className=""
                        onClick={[Function]}
                />
        </div>
        <p>
                Model: 
                Digger Crawler
        </p>
        <p>
                Class: 
                wheeled
        </p>
        <p>
                Passengers: 
                50
        </p>
</div>,
      "className": "info-card",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="info-card__top-line"
>
            <h3>
                        Sand Crawler
            </h3>
            <button
                        className=""
                        onClick={[Function]}
            />
</div>,
          <p>
            Model: 
            Digger Crawler
</p>,
          <p>
            Class: 
            wheeled
</p>,
          <p>
            Passengers: 
            50
</p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h3>
                Sand Crawler
</h3>,
              <button
                className=""
                onClick={[Function]}
/>,
            ],
            "className": "info-card__top-line",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sand Crawler",
              },
              "ref": null,
              "rendered": "Sand Crawler",
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Model: ",
              "Digger Crawler",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Model: ",
            "Digger Crawler",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Class: ",
              "wheeled",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Class: ",
            "wheeled",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Passengers: ",
              "50",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Passengers: ",
            "50",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <div
                    className="info-card__top-line"
          >
                    <h3>
                              Sand Crawler
                    </h3>
                    <button
                              className=""
                              onClick={[Function]}
                    />
          </div>
          <p>
                    Model: 
                    Digger Crawler
          </p>
          <p>
                    Class: 
                    wheeled
          </p>
          <p>
                    Passengers: 
                    50
          </p>
</div>,
        "className": "info-card",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="info-card__top-line"
>
              <h3>
                            Sand Crawler
              </h3>
              <button
                            className=""
                            onClick={[Function]}
              />
</div>,
            <p>
              Model: 
              Digger Crawler
</p>,
            <p>
              Class: 
              wheeled
</p>,
            <p>
              Passengers: 
              50
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h3>
                  Sand Crawler
</h3>,
                <button
                  className=""
                  onClick={[Function]}
/>,
              ],
              "className": "info-card__top-line",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sand Crawler",
                },
                "ref": null,
                "rendered": "Sand Crawler",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "button",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Model: ",
                "Digger Crawler",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Model: ",
              "Digger Crawler",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Class: ",
                "wheeled",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Class: ",
              "wheeled",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Passengers: ",
                "50",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Passengers: ",
              "50",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
